taichi/backends/vulkan/vulkan_loader.cpp

```diff
@@ -11,6 +11,80 @@ namespace vulkan {
 VulkanLoader::VulkanLoader() {
 }
 
+bool VulkanLoader::check_vulkan_device() {
+  bool found_device_with_compute = false;
+
+  // We create an temporary Vulkan instance to probe the Vulkan devices.
+  // Otherwise, in the case of a CPU only VM with Vulkan installed, Vulkan will
+  // not run as there is no GPU available, but the fallback will not happen
+  // because Vulkan API is available.
+
+  VkApplicationInfo app_info{};
+  app_info.sType = VK_STRUCTURE_TYPE_APPLICATION_INFO;
+  app_info.pApplicationName = "Checking Vulkan Device";
+  app_info.applicationVersion = VK_MAKE_VERSION(1, 0, 0);
+  app_info.pEngineName = "No Engine";
+  app_info.engineVersion = VK_MAKE_VERSION(1, 0, 0);
+  app_info.apiVersion = VK_API_VERSION_1_0;
+
+  VkInstanceCreateInfo create_info{};
+  create_info.sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO;
+  create_info.pApplicationInfo = &app_info;
+
+  VkInstance instance{VK_NULL_HANDLE};
+  VkResult res = vkCreateInstance(&create_info, kNoVkAllocCallbacks, &instance);
+
+  do {
+    if (res != VK_SUCCESS) {
+      TI_WARN("Can not create Vulkan instance");
+      break;
+    }
+
+    load_instance(instance);
+
+    uint32_t device_count = 0;
+    vkEnumeratePhysicalDevices(instance, &device_count, nullptr);
+
+    if (device_count == 0) {
+      TI_WARN("Can not find Vulkan capable devices");
+      break;
+    }
+
+    std::vector<VkPhysicalDevice> devices(device_count);
+    vkEnumeratePhysicalDevices(instance, &device_count, devices.data());
+
+    for (int i = 0; i < devices.size(); i++) {
+      const auto &physical_device = devices[i];
+
+      uint32_t queue_family_count = 0;
+      vkGetPhysicalDeviceQueueFamilyProperties(physical_device,
+                                               &queue_family_count, nullptr);
+      if (queue_family_count > 0) {
+        std::vector<VkQueueFamilyProperties> queue_families(queue_family_count);
+        vkGetPhysicalDeviceQueueFamilyProperties(
+            physical_device, &queue_family_count, queue_families.data());
+
+        for (auto &queue : queue_families) {
+          if (queue.queueFlags & VK_QUEUE_COMPUTE_BIT) {
+            found_device_with_compute = true;
+          }
+        }
+
+        VkPhysicalDeviceProperties properties{};
+        vkGetPhysicalDeviceProperties(physical_device, &properties);
+
+        TI_INFO("Found Vulkan Device {} ({})", i, properties.deviceName);
```

Comment:
```suggestion
        TI_DEBUG("Found Vulkan Device {} ({})", i, properties.deviceName);
```

Reply:
This is in conjunction with #4255 so that the user can know which device corresponds to which ID

Reply:
Maybe is there a way to display info without including the source file path?